.github/workflows/go-build.yml

```diff
@@ -0,0 +1,21 @@
+name: Build
+
+on:
+  push:
+    branches: [ "main", "develop" ]
```

Comment:
maybe we need this for the `pull_request`?. the release job right now taking care of building for main when we push new tag

Reply:
i will merge and change it and add test execution